workshops/models.py

```diff
@@ -296,17 +296,20 @@ def get_by_ident(ident):
         YYYY-MM-DD-site slug, and indicates an event whose dates are firm.
         The real indicator is the 'published' flag.
         '''
-
-        # match event ID
-        if re.match(r'^\d+$', ident):
-            return Event.objects.get(id=ident)
-
-        # match event slug in format YYYY-MM-**** (ie. this works for
-        # YYYY-MM-DD-**** too)
-        if re.match(r'^\d{4}-\d{2}-.+$', ident):
-            return Event.objects.get(slug=ident)
-
-        raise ObjectDoesNotExist(ident)
+        kwargs = {}
+        try:
+            # first try to match using ID
+            kwargs["id"] = int(ident)
+        except ValueError:
+            # match event slug in format YYYY-MM-****
+            #(ie. this works for YYYY-MM-DD-**** too)
+            if re.match(r'^\d{4}-\d{2}-.+$', ident):
+                kwargs["slug"] = ident
+            else:
+                raise ObjectDoesNotExist(ident)
+
+        # may throw DoesNotExist exception
+        return Event.objects.get(**kwargs)
```

Comment:
On Mon, Apr 20, 2015 at 10:22:14PM -0700, Piotr Banaszkiewicz wrote:
> Or we can allow any slug but integer by matching `\d+` in
> `EventForm.clean_slug`.
>
> Personally, the combo "suggestive help_text" + "not allowing integer
> slugs" appeals the most to me. But I'll be fine even if we leave it
> as it is right now.

So raise a ValidationError if we can match the slug with an
integer-only regexp [1]?

[1]: https://docs.djangoproject.com/en/1.8/ref/forms/validation/